core/src/main/java/io/questdb/griffin/UpdateOperatorImpl.java

```diff
@@ -144,6 +144,14 @@ public long executeUpdate(SqlExecutionContext sqlExecutionContext, UpdateOperati
                     final int rowPartitionIndex = Rows.toPartitionIndex(rowId);
                     final long currentRow = Rows.toLocalRowID(rowId);
 
+                    if (rowPartitionIndex > -1 && tableWriter.isPartitionReadOnly(rowPartitionIndex)) {
```

Comment:
Please move this check inside the following `if` to not execute it on every row but on the partition index change only.